### PR TITLE
fix(client): Only require `signalArgs` when needed

### PR DESCRIPTION
**Pull request by lorensr**
when calling `signalWithStart`

### CODE REVIEW

File: packages/client/src/workflow-options.ts [hunk 1]
```diff
@@ -37,15 +37,26 @@ export interface WorkflowOptions extends CommonWorkflowOptions {
...
   /**
    * SignalDefinition or name of signal
    */
-  signal: SignalDefinition<SA> | string;
+  signal: SignalDefinition<SignalArgs> | string;
   /**
    * Arguments to invoke the signal handler with
    */
-  signalArgs: SA;
+  signalArgs: SignalArgs;
```
Comment (lorensr) [inline packages/client/src/workflow-options.ts:59]: If there's a way for an interface to have a conditional property, I couldn't find it 😄

File: packages/client/src/workflow-options.ts [hunk 2]
```diff
@@ -37,15 +37,26 @@ export interface WorkflowOptions extends CommonWorkflowOptions {
   followRuns?: boolean;
 }
 
-export interface WorkflowSignalWithStartOptions<SA extends any[] = []> extends WorkflowOptions {
+export type WorkflowSignalWithStartOptions<SignalArgs extends any[] = []> = SignalArgs extends []
```
Comment (bergundy) [inline packages/client/src/workflow-options.ts:40]: You want the condition to be the other way around.
See https://github.com/temporalio/sdk-typescript/blob/a7be2cd12bf535df607a53035373e33d74c28e25/packages/internal-workflow-common/src/workflow-options.ts#L87

Comment (lorensr) [reply packages/client/src/workflow-options.ts]: Ah, I thought `[]` would just be empty array. I've changed to an array with at least one item:`[any, ...any[]]`

File: packages/client/src/workflow-options.ts [hunk 3]
```diff
@@ -37,15 +37,26 @@ export interface WorkflowOptions extends CommonWorkflowOptions {
...
-export interface WorkflowSignalWithStartOptions<SA extends any[] = []> extends WorkflowOptions {
+export type WorkflowSignalWithStartOptions<SignalArgs extends any[] = []> = SignalArgs extends [any, ...any[]]
+  ? WorkflowSignalWithStartOptionsWithArgs<SignalArgs>
+  : WorkflowSignalWithStartOptionsWithoutArgs;
+
+export interface WorkflowSignalWithStartOptionsWithoutArgs extends WorkflowOptions {
+  /**
+   * SignalDefinition or name of signal
+   */
+  signal: SignalDefinition | string;
```
Comment (bergundy) [inline packages/client/src/workflow-options.ts:48]: I'd support optional empty args here, same way we do for regular workflow `args`

Comment (lorensr) [reply packages/client/src/workflow-options.ts:48]: Thanks, updated

**Review by bergundy — APPROVED** [review]
LGTM, thanks

**Pull request merged**